tasks/hough-transform/src/main.rs

```diff
@@ -41,14 +42,17 @@ fn load_pgm(filename: &str) -> ImageGray8 {
     let mut img = ImageGray8 {
         width,
         height,
-        data: repeat(0u8).take(width * height).collect(),
+        data: vec![],
     };
 
     // Read image data
-    let len = img.data.len();
-    match file.read(&mut img.data) {
```

Comment:
This function should return `io::Result<ImageGray8>` and we should replace all the unwraps with error propagation. For example:

```rust
let expected_bytes = width * height;
let bytes_read = file.read_to_end(&mut img.data)?;
if bytes_read != expected_bytes {
  let kind = bytes_read < expected_bytes { io::ErrorKind::UnexpectedEof } else { io::ErrorKind::InvalidData };
  let msg = format!("expected {} bytes", expected_bytes);
  return Err(io::Error::new(kind, msg));
}
```

Reply:
I just want to fix the transformation, but I agree that the error handling in the original code was not very robust. See the latest commit I have reworked this function to propagate errors.